File: Leetcode 3101-3200/3179/3179.java
// Leetcode 3179: Find the N-th Value After K Seconds
// https://leetcode.com/problems/find-the-nth-value-after-k-seconds/
// Solved on 8th of December, 2025
class Solutio {
    /**
     * Calculates the value of the n-th element after k seconds.
     * @param n The number of elements.
     * @param k The number of seconds.
     * @return The value of the n-th element after k seconds, modulo 10^9 + 7.
     */
    public int valueAfterKSeconds(int n, int k) {
        int mod = 1000000007;
        int[] dp = new int[n];
        for (int i = 0; i < n; i++) {
            dp[i] = 1;
        }
        for (int i = 0; i < k; i++) {
            for (int j = 1; j < n; j++) {
                dp[j] = (dp[j] + dp[j - 1]) % mod;
            }
        }
        return dp[n - 1];
    }
}

File: Leetcode 3101-3200/3179/3179.py
# Leetcode 3179: Find the N-th Value After K Seconds
# https://leetcode.com/problems/find-the-n-th-value-after-k-seconds/
# Solved on 8th of December, 2025
class Solution:
    def valueAfterKSeconds(self, n: int, k: int) -> int:
        """
        Calculates the n-th value after k seconds.

        Args:
            n (int): The number of elements.
            k (int): The number of seconds.

        Returns:
            int: The n-th value after k seconds, modulo 10^9 + 7.
        """
        modulo = 10**9 + 7
        result = 1
        for i in range(n - 1):
            result *= (k + 1 + i)
            result //= (i + 1)

        return result % modulo